test/e2e/parallel/space_test.go

```diff
@@ -498,19 +498,27 @@ func TestSubSpaceInheritance(t *testing.T) {
 		// when
 		// we have a parentSpace
 		t.Logf("Create parent space")
-		parentSpace, _, parentSpaceBindings := CreateSpace(t, awaitilities, testspace.WithSpecTargetCluster(memberAwait.ClusterName), testspace.WithTierName("appstudio"))
-		// then
-		// wait until MUR and Space have been provisioned
-		t.Logf("Wait for space to be provisioned")
-		VerifyResourcesProvisionedForSpace(t, awaitilities, parentSpace.Name, UntilSpaceHasStatusTargetCluster(memberAwait.ClusterName))
-		_, err = hostAwait.WaitForMasterUserRecord(t, parentSpaceBindings.Spec.MasterUserRecord)
-		require.NoError(t, err)
+		parentUser := NewSignupRequest(awaitilities).
+			ManuallyApprove().
+			TargetCluster(memberAwait).
+			RequireConditions(ConditionSet(Default(), ApprovedByAdmin())...).
+			SpaceTier("appstudio").
+			EnsureMUR().
+			Execute(t)
 
 		// when
 		// we also have a subSpace with same tier but with disable inheritance
 		t.Logf("Create sub space with role: contributor")
-		subSpace, _, subSpaceBindings := CreateSpaceWithRole(t, awaitilities, "contributor",
-			testspace.WithSpecParentSpace(parentSpace.Name),
+		subUser := NewSignupRequest(awaitilities).
+			ManuallyApprove().
+			RequireConditions(ConditionSet(Default(), ApprovedByAdmin())...).
+			NoSpace().
+			WaitForMUR().
+			Execute(t)
```

Comment:
correct me if I'm wrong, but we should not replace the creation of the sub-space with provisioning via UserSignup. In this case, we should really create directly the Space CR and we don't have to worry about the cleanup since the automatic cleanup is disabled for sub-spaces

Reply:
So sorry!! My bad! Thanks for noticing it!

Reply:
@MatousJobanek, should I rollback also the change in `CreateSpaceWithBinding` since we do not need to specify the role, or we can leave it this way for future proposes?

Reply:
@MatousJobanek, I was checking now and `CreateSpaceWithRole` creates via UserSignup

Reply:
> should I rollback also the change in CreateSpaceWithBinding since we do not need to specify the role, or we can leave it this way for future proposes?

keep it as it is - it looks fine.

> I was checking now and CreateSpaceWithRole creates via UserSignup

good catch. 
The test verifies that we can create a sub-space with disabled inheritance (the SpaceBindings from the parent space are not inherited) and share it with a user. As a result, only the RoleBindings for user the sub-space was shared with will be created, and not for users that has access to the parent space. Technically speaking, provisioning the Space via a UserSignup would do the same, but it's more about how the test is written.

Reply:
Thank you so much! I addressed the creation of the space directly